lib/import/colorectal/providers/newcastle/newcastle_handler_colorectal.rb

```diff
@@ -7,341 +7,230 @@ module Providers
+            gene = record.raw_fields['gene']
+            positive_genes = gene.scan(COLORECTAL_GENES_REGEX).flatten.uniq unless gene.nil?
+            if positive_genes.size.zero?
+              positive_genes = record.raw_fields['investigation code'].
+                               scan(COLORECTAL_GENES_REGEX).flatten.uniq
             end
-            genotypes
+            positive_genes[0] unless positive_genes.nil?
+          end
+
+          def full_screen?(genocolorectal)
+            return false if genocolorectal.attribute_map['genetictestscope'].nil?
+
+            genocolorectal.attribute_map['genetictestscope'].scan(/Full screen/i).size.positive?
+          end
+
+          def targeted?(genocolorectal)
+            return false if genocolorectal.attribute_map['genetictestscope'].nil?
+
+            genocolorectal.attribute_map['genetictestscope'].scan(/Targeted/i).size.positive?
           end
 
-          def summarize
-            @logger.info ' ************** Handler Summary **************** '
-            @logger.info "Num bad variants: #{@failed_variant_counter} of "\
-                         "#{@variants_processed_counter} processed"
+          def no_scope?(genocolorectal)
+            return false if genocolorectal.attribute_map['genetictestscope'].nil?
+
+            genocolorectal.attribute_map['genetictestscope'].scan(/Unable/i).size.positive?
           end
+
+          def pathogenic?(record)
+            varpathclass = record.raw_fields['variantpathclass']&.downcase
+            return true if NON_PATHEGENIC_CODES.exclude? varpathclass
+
+            false
+          end
+
+          def process_variants(genocolorectal, variant)
+            process_cdna_variant(genocolorectal, variant)
+            process_exonic_variant(genocolorectal, variant)
+            process_protein_impact(genocolorectal, variant)
+          end
+
+          def process_exonic_variant(genocolorectal, variant)
+            return unless variant.scan(EXON_VARIANT_REGEX).size.positive?
+
+            genocolorectal.add_exon_location($LAST_MATCH_INFO[:exons])
+            genocolorectal.add_variant_type($LAST_MATCH_INFO[:variant])
+            @logger.debug "SUCCESSFUL exon variant parse for: #{variant}"
+          end
+
+          def process_cdna_variant(genocolorectal, variant)
+            return unless variant.scan(CDNA_REGEX).size.positive?
+
+            genocolorectal.add_gene_location($LAST_MATCH_INFO[:cdna])
+            @logger.debug "SUCCESSFUL cdna change parse for: #{$LAST_MATCH_INFO[:cdna]}"
+          end
+
+          def process_protein_impact(genocolorectal, variant)
+            if variant.scan(PROTEIN_REGEX).size.positive?
+              genocolorectal.add_protein_impact($LAST_MATCH_INFO[:impact])
+              @logger.debug "SUCCESSFUL protein parse for: #{$LAST_MATCH_INFO[:impact]}"
+            else
+              @logger.debug "FAILED protein parse for: #{variant}"
+            end
+          end
+
```

Comment:
Remove the blank line please :)

Reply:
Removed it and rubocopped the whole file.